corehq/apps/data_interfaces/migrations/0036_backfill_dedupe_match_values.py

```diff
@@ -29,24 +31,17 @@ def process_records(batch_size=100, db_alias=None):
             CaseDuplicateNew.objects.bulk_create(inserts, ignore_conflicts=True)
 
 
-# Copied from: https://djangosnippets.org/snippets/1170/
-def batch_qs(qs, batch_size=1000):
-    """
-    Returns a (start, end, total, queryset) tuple for each batch in the given
-    queryset.
-
-    Usage:
-        # Make sure to order your querset
-        article_qs = Article.objects.order_by('id')
-        for start, end, total, qs in batch_qs(article_qs):
-            print "Now processing %s - %s of %s" % (start + 1, end, total)
-            for article in qs:
-                print article.body
-    """
-    total = qs.count()
-    for start in range(0, total, batch_size):
-        end = min(start + batch_size, total)
-        yield (start, end, total, qs[start:end])
+def batch_query(db_alias, model, query, batch_size):
+    it = paginate_query(db_alias, model, query, query_size=batch_size)
+    return batched(it, batch_size)
+
+
+def batched(iterable, n):
```

Comment:
nit: We have a [`chunked`](https://github.com/dimagi/commcare-hq/blob/1dbc6976eb41f912af08bd187111c24374f8a500/corehq/ex-submodules/dimagi/utils/chunked.py#L4) utility that does this.